app/main/page.tsx

```diff
@@ -44,29 +36,53 @@ function App() {
   const [nameTagContent, setNameTagContent] =
     useState<NameTagContent>(defaultNameTag);
 
-  const [nameTagIsLoaded, setNameTagIsLoaded] = useState(false);
+  const [waveHandButtons, setWaveHandButtons] = useState<string[]>([]);
+  const [hasError, setHasError] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const foregroundDrawerRef = useRef<DrawBadgeApi | null>(null);
+
+  useEffect(() => {
+    const init = async () => {
+      const zoomApi: ZoomApiWrapper = await getZoomApi();
+      foregroundDrawerRef.current = new DrawBadgeApi(zoomApi);
+    };
+    init();
+  }, []);
 
   const updateNameTagContent: SubmitHandler<NameTagContent> = (data) => {
     if (nameTagContent.visible !== data.visible) {
       log(data.visible ? Action.NAME_BADGE_ON : Action.NAME_BADGE_OFF);
     }
     setNameTagContent(data);
-    foregroundDrawer.drawNameTag(data);
+    foregroundDrawerRef.current?.drawNameTag(data);
   };
 
   const updateHandWaveBadge = (badge: HandWaveBadge) => {
-    foregroundDrawer.drawHandWave(badge);
+    foregroundDrawerRef.current?.drawHandWave(badge);
   };
 
-  useEffect(() => {
-    fetchNametagFromDB()
-      .then((newNameTag) => {
-        if (newNameTag !== undefined) {
-          setNameTagContent(newNameTag);
+  const fetchUser = async () => {
+    fetchUserFromDB()
+      .then((user) => {
+        if (user.nameTag !== undefined) {
+          setNameTagContent(user.nameTag);
+        }
```

Comment:
This can be written equivalently like:

```suggestion
    const user = await fetchUserFromDB();
    if (user.nameTag !== undefined) {
      setNameTagContent(user.nameTag);
    }
    ...
```

`.catch` chains can be equivalently written as `try-catch` blocks.